core/api/src/main/java/com/microsoft/gctoolkit/time/DateTimeStamp.java

```diff
@@ -252,4 +256,11 @@ public double timeSpanInMinutes(DateTimeStamp dateTimeStamp) {
         return this.minus(dateTimeStamp) / 60.0d;
     }
 
+    @Override
+    public int compareTo(DateTimeStamp dateTimeStamp) {
+        return  comparator.compare(this,dateTimeStamp);
+    }
+    private static  Comparator<DateTimeStamp> getComparator(){
+        return nullsLast(comparing(DateTimeStamp::getDateTime));
```

Comment:
DateTimeStamp is kind of a tricky thing because it depends on the log file flags. Sometimes, there is no ISO date/time in the log. Sometimes there isn't even a time stamp (which represents milliseconds since the start of the JVM). Indeed, most of the time we only care about the time stamp. 

The way this class is written and used is to try to ensure there is always a timeStamp. It is possible that timeStamp could be NaN, or less than zero - neither of which make sense and point to a bug in the code that calls the constructor. 

So you may or may not have a dateTime, but you should always have a timeStamp >= 0. One possible way to do this compare is to compare the dateTime (if not null) and if they are the same, then compare the timeStamp. Another possible way would be to add the timeStamp value to the dateTime (if not null) and compare that.